### PR TITLE
Ladybird: Abstract spawning helper processes into separate methods

### LINKED ISSUE #17062: Ladybird always looks for SQLServer in ./SQLServer/SQLServer, does not look in CMAKE_INSTALL_BINDIR
In this line

https://github.com/SerenityOS/serenity/blob/f15aa539be22d87fa4bb65ff641e5ebae4027d91/Ladybird/main.cpp#L58

ladybird always executes `./SQLServer/SQLServer`, but if ladybird is installed (I'm trying this in a linux system), it should look for it in `$CMAKE_INSTALL_BINDIR`

### PR DESCRIPTION
**Pull request by ADKaster**
This will let us use the same path discovery methods for WebContent,
SQLServer, and any other helper processes we need to launch.

Fixes #17062

### CODE REVIEW

File: Ladybird/HelperProcess.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        if (!result.is_error())
+            break;
+    }
+
+    return result;
+}
+
+ErrorOr<Vector<String>> get_paths_for_helper_process(StringView process_name)
+{
+    Vector<String, 2> paths;
```
Comment (trflynn89) [inline Ladybird/HelperProcess.cpp:29]: Seems odd to have an inline storage of 2, and then add 3 paths? Probably fine to just not have inline storage at all here

**Review by trflynn89 — COMMENTED** [review]
One minor nit, lgtm though

File: Ladybird/HelperProcess.cpp (hunk 1 above)
Comment (ADKaster) [reply Ladybird/HelperProcess.cpp]: Good point. I previously had 2 paths here and it made a bit more sense. removed the inline capacity.

File: Ladybird/WebDriver/main.cpp [hunk 2]
```diff
@@ -33,6 +35,25 @@ static char** environment()
 #endif
 }
 
+static ErrorOr<pid_t> launch_process(StringView application, char const* argv[])
+{
+    // FIXME: Why can't we TRY() an ErrorOr<T> in a function returning ErrorOr<pid_t>?
```
Comment (trflynn89) [inline Ladybird/WebDriver/main.cpp:40]: hmm what error did you get here? I pulled this locally and was able to build with:
```diff
-    // FIXME: Why can't we TRY() an ErrorOr<T> in a function returning ErrorOr<pid_t>?
-    auto paths_or_error = get_paths_for_helper_process(application);
-    if (paths_or_error.is_error())
-        return paths_or_error.release_error();
-    auto paths = paths_or_error.release_value();
+    auto paths = TRY(get_paths_for_helper_process(application));
```

Comment (ADKaster) [reply Ladybird/WebDriver/main.cpp:40]: ... huh. It was complaining about no constructor to convert b/w the TRY'd type and `ErrorOr<pid_t>`. But you're right, that works now. ... I am confused but happy. I'll push that then.

**Pull request merged**